scripts/test/Muon/PeriodicTableWidget_test.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import absolute_import, print_function
+
+import unittest
+
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_view import PeriodicTableView
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_presenter import PeriodicTablePresenter
+from Muon.GUI.ElementalAnalysis.PeriodicTable.periodic_table_widget import PeriodicTable
+from Muon.GUI.Common import mock_widget
+
+from PyQt4 import QtGui
+
+from silx.gui.widgets import PeriodicTable as silxPT
+
+from silx.gui.widgets.PeriodicTable import ColoredPeriodicTableItem
+
+try:
+    from unittest import mock
+except ImportError:
+    import mock
+
+
+class MockEvent(object):
+    def __init__(self):
+        self.call_count = 0
+
+    def handler(self):
+        self.call_count += 1
+
+
+class PeriodicTableWidgetTest(unittest.TestCase):
+    def setUp(self):
+        self._qapp = mock_widget.mockQapp()
+        self.widget = PeriodicTable()
+        self.view = self.widget.presenter.view
+        self.mock_elem = ColoredPeriodicTableItem("T", 0, "", 1, "Test", 0)
+        self.mock_symbol = "T"
+
+        self.view.ptable = mock.create_autospec(silxPT)
+        self.view.ptable.getSelection = mock.Mock(
+            return_value=self.mock_symbol)
+        self.view.ptable.setSelection = mock.Mock()
+        self.view.ptable.isElementSelected = mock.Mock(return_value=True)
+        self.view.ptable.setElementSelected = mock.Mock()
+
+    def call_event_once(self, reg_event, func, args,
```

Comment:
I find this function unclear

Reply:
The setUp() or call_event_once()? Guessing the former but they're both highlighted.

Reply:
`call_event_once`